lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -373,27 +374,97 @@ def send_request_raw(opts = {}, timeout = 20, disconnect = false)
   # Connects to the server, creates a request, sends the request,
   # reads the response
   #
+  # If a +HTTP::CookieJar+ instance is passed in the +opts+ dict under a 'cookie' key, said CookieJar will be used in
+  # the request instead of the module +cookie_jar+
+  #
   # Passes `opts` through directly to {Rex::Proto::Http::Client#request_cgi}.
   # Set `opts['keep_cookies']` to keep cookies from responses for reuse in requests.
+  # Cookies returned by the server will be stored in +cookie_jar+
   #
   # @return (see Rex::Proto::Http::Client#send_recv))
   def send_request_cgi(opts = {}, timeout = 20, disconnect = true)
-    if cookie_jar.any?
-      opts = { 'cookie' => cookie_jar.to_a.join(' ') }.merge(opts)
+    url = "http#{ssl ? 's' : ''}://#{peer}"
```

Comment:
is this the expected format of the url? I was pretty sure a cookies domain was something like `.github.com` or `github.com` without the protocol?

Reply:
Yeah I think I got over zealous here, I'll double check if the protocol can be removed since I'm pretty sure your right

Reply:
Seems like a full URL is needed for `!cookie_jar.empty?(url)` & `cookie_jar.cookies(url)`, so the current URL format should be kept